exodus_lambda/functions/origin_request.py

```diff
@@ -103,11 +104,11 @@ def handler(self, event, context):
                 )
                 content_type = query_result["Items"][0]["content_type"]["S"]
                 if content_type:
-                    query = urllib.parse.urlencode(
-                        {"ResponseContentType": content_type}
-                    )
-                    request["uri"] += "?" + query
+                    request[
+                        "querystring"
+                    ] = "response-content-type={0}".format(content_type)
```

Comment:
The encoding breaks the functionality. The tests are failing as soon as it is restored.

Reply:
If we're worried about slashes in the querystring...
> urllib.parse.quote(content_type, safe="")
'text%2Fplain'

But if integration tests pass without URL encoding, slashes _must_ be safe. I feel that AWS should accept encoded as well, though, so failures due to this don't make sense to me.

Reply:
Are you sure encryption is required? I have reviewed the documentation again. Didn't find any reference to encryption for this one.

Reply:
I'm not sure it's required.  I realized later that it may or may not be required because cloudfront might be decoding the query before it puts it into 'querystring', or it might not... and the docs are completely silent on the point.

It doesn't make sense to me that encoding wouldn't be needed in general though, seeing as it comes as a single string.  If you imagine for instance that the client wanted to pass the equivalent of `{'snacks': 'peanuts&chocolate', 'meals': 'sandwich'}` in their request, logically it *must* be coming into the lambda with that & encoded to %26, i.e. `meals=sandwich&snacks=peanuts%26chocolate`, otherwise there would be no safe way to use the query string.  So I conclude that either values in querystring *are* meant to be encoded, or else the lambda event design is majorly flawed.

Anyway, due to lack of details in the docs, it seems we just need to accept whatever works.

Reply:
Yeah, I think the reason encoding the `content_type` right now doesn't seem necessary is because only strings containing slashes are tested. Slashes may be safe, then, but we can still encode the `content_type` to protect against other special characters that may be introduced in the future.
```suggestion
                    ] = "response-content-type={0}".format(urllib.parse.quote(content_type))
```
This shouldn't change the current behavior, "text/plain" is still "text/plain".